src/dist/mod.rs

```diff
@@ -1166,9 +1169,20 @@ pub(crate) async fn dl_v2_manifest(
             Ok(Some((manifest, manifest_hash)))
         }
         Err(any) => {
-            if let Some(RustupError::ChecksumFailed { .. }) = any.downcast_ref::<RustupError>() {
-                // Checksum failed - issue warning to try again later
-                info!("update not yet available, sorry! try again later")
+            if let Some(err @ RustupError::ChecksumFailed { .. }) =
+                any.downcast_ref::<RustupError>()
+            {
+                // Manifest checksum mismatched.
+                warn!("{err}");
+
+                let server = dist_root_server(download.process)?;
+                if server == DEFAULT_DIST_SERVER {
+                    // https://github.com/rust-lang/rustup/issues/3390
+                    info!("this is likely due to an ongoing update of the official release server, please try again later");
```

Comment:
Maybe add these links to the actual output? At least for I always like the chance to better understand the details of why a problem happens.